File: arkts_static_test/03_Types/ets_cases/3.17.1_Readonly_Tuple_Types/cross_lang_verify/compile-fail/TYP_03_17_01_004.java
/**
 * Java cross-language verification for TYP_03_17_01_004_FAIL_INDEX_WRITE
 * Java final 数组的元素仍然可以修改
 */
class TYP_03_17_01_004 {
    public static void main(String[] args) {
        final Object[] tuple = {1, "abc"};

        // Java: final 数组的元素可以修改（只有引用不可变）
        tuple[0] = 42;
        System.out.println("Java: final array elements can be modified: " + tuple[0]);

        System.out.println("verified");
    }
}

File: arkts_static_test/03_Types/ets_cases/3.17.1_Readonly_Tuple_Types/cross_lang_verify/compile-fail/TYP_03_17_01_004.swift
/**
 * Swift cross-language verification for TYP_03_17_01_004_FAIL_INDEX_WRITE
 * Swift let 元组的元素不能修改
 */
import Foundation

let tuple: (Int, String) = (1, "abc")

// Swift: let 元组的元素不能修改
// tuple.0 = 42  // 编译错误: Cannot assign to property: 'tuple' is a 'let' constant

print("Swift: let tuple elements cannot be modified")
print("verified")
